10-std.tex

```diff
@@ -539,3 +539,145 @@ \section{Input/Output}
+
+This client code example connects to the server remoting context, and calls a function \expr{foo()} on its API.
+\begin{lstlisting}
+class Client {
+  static function main() {
+    var cnx = haxe.remoting.HttpAsyncConnection.urlConnect("http://localhost/");
+    cnx.setErrorHandler( function(err) trace('Error: $err'); } );
+    cnx.Server.foo.call([1,2], function(data) trace('Result: $data'););
+  }
+}
+\end{lstlisting}
+
+To make this work for Neko target, setup a Neko Web Server, point the url in the Client to \ic{"http://localhost2000/remoting.n"} and compile the Server using \ic{-main Server -neko remoting.n}.
+
+\paragraph{Error handling}
+
+\begin{itemize}
+	\item When an error occurs in a asynchronous call, the error handler is called as seen in this example above.
+	\item When an error occurs in a synchronous call, an exception is raised on the caller-side, as if we were calling a local method.
+\end{itemize}
+
+\paragraph{Data serialization}
+
+Haxe Remoting can send a lot of different kinds of data. See \tref{Serialization}{std-serialization}.
+
+\subsection{Implementation details}
+\label{std-remoting-implementation-details}
+
+\paragraph{Javascript security specifics}
+
+The html-page wrapping the js client must be served from the same domain as where the server is running. The same-origin policy restricts how a document or script loaded from one origin can interact with a resource from another origin. Same-origin Policy is used as a means to prevent some of the Cross-site Request Forgery attacks.
+
+To use the remoting across domain boundaries, CORS (Cross-Origin Resource Sharing) needs to be enabled by defining the header \ic{X-Haxe-Remoting} in the \ic{.htaccess}:
+
+\begin{lstlisting} 
+# Enable CORS
+Header set Access-Control-Allow-Origin "*"
+Header set Access-Control-Allow-Methods: "GET,POST,OPTIONS,DELETE,PUT"
+Header set Access-Control-Allow-Headers: X-Haxe-Remoting
+\end{lstlisting} 
+
+See \href{http://en.wikipedia.org/wiki/Same-origin_policy}{same-origin policy} for more information on this topic.
+
+Also note that this means that the page can't be served directly from the file system \ic{"file:///C:/example/path/index.html"}.
+
+\paragraph{Flash security specifics}
+
+When Flash accesses a server from a different domain, set up a \ic{crossdomain.xml} file on the server, enabling the \ic{X-Haxe} headers.
+
+\begin{lstlisting} 
+<cross-domain-policy>
+	<allow-access-from domain="*"/> <!-- or the appropriate domains -->
+	<allow-http-request-headers-from domain="*" headers="X-Haxe*"/>
+</cross-domain-policy>
+\end{lstlisting} 
+
+\paragraph{Arguments types are not ensured}
+
+There is no warranty of any kind that the arguments types will be respected when a method is called using remoting. 
+That means even the arguments of function \expr{foo} are typed to \type{Int}, the client will still be able to use strings while calling the method. 
```

Comment:
Did you mean to use "That means even if..."? otherwise please review the sentence